pypif_sdk/func/calculate_funcs.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*- 
+from __future__ import unicode_literals
+import pypif
+import re
+from pypif.obj import *
+import json
+import os.path
+
+path = os.path.abspath(os.path.dirname(__file__))
+path_to_json = os.path.join(path, "elements.json")
+with open(path_to_json) as f:
+    elements_data = json.load(f)
+
+def calculate_ideal_atomic_percent(pif):
+    """
+    Calculates ideal atomic percents from a chemical formula string from a pif. Returns an appended pif with composition elements modified or added.
+    :param - pif a ChemicalSystem pif
+    returns - modified pif object
+    """
+    try:
+        pif.chemicalFormula
```

Comment:
The ChemicalSystem doesn't have a `chemicalFormula` attribute; it should be `chemical_formula` (see [here](https://github.com/CitrineInformatics/pypif/blob/develop/pypif/obj/system/chemical/chemical_system.py#L12)).